### PR TITLE
CLOUD-1076 test am metadata

### PR DESCRIPTION
**Pull request by andrewvinall**
Jira issue? <CLOUD-1076>
Release 6.5.0 backport required? <no>
6.5.0 doc changes needed? <no>
7.0.0 doc changes needed? <no>
Required README updates made? <no>

POC for checking data about the AM and Amster images e.g. version, jdk, presence of legal-notices.
An example manifest file is attached:
[6.5.0-manifest.txt](https://github.com/ForgeRock/forgeops/files/2822330/6.5.0-manifest.txt)

### CODE REVIEW

File: cicd/forgeops-tests/lib/utils/kubectl.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# Common Development and Distribution License (CDDL) that can be found in the LICENSE file
+
+# Lib imports
+import subprocess
+
+# Framework imports
+from utils import logger
+
+KUBECTL_COMMAND = "kubectl"
+
```
Comment (wstrange) [inline cicd/forgeops-tests/lib/utils/kubectl.py]: You guys prolly know this - but there is a native python library for talking to k8s
No need for kubectl

Comment (wstrange) [reply cicd/forgeops-tests/lib/utils/kubectl.py]: yep - that is the one. It might be more robust / easier vs. shelling to kubectl

**Comment by andrewvinall** [comment]
I have refactored/rewritten the code to in attempt to simplify the tests and improve the structure.
The kubernetes Python library is now being used, but not for `exec` as I couldn't get this to work.

Please note this snippet:
```
class TestAMMetadata(object):
    MANIFEST_FILEPATH = "/tmp/6.5.0-manifest.txt" # TODO: Currently this hard code blocks this change

File: cicd/forgeops-tests/lib/utils/am_pod.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+"""
+
+# Lib imports
+import os
+from requests import get, post
+
+# Framework imports
+from ProductConfig import AMConfig
+from utils import logger, rest, kubectl
+from utils.pod import Pod
```
[the remark this answers is not in the document]
Comment (andrewvinall) [reply cicd/forgeops-tests/lib/utils/am_pod.py]: cicd/forgeops-tests/lib/utils/pod.py

File: cicd/forgeops-tests/lib/utils/pod.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    @property
+    def config(self):
+        return self._config
+
+    @property
+    def manifest(self):
+        return self._manifest
+
+    @staticmethod
+    def is_expected_am_amster_version(am_manifest, amster_manifest):
```
[the remark this answers is not in the document]
Comment (andrewvinall) [reply cicd/forgeops-tests/lib/utils/pod.py]: The test compares AM and Amster versions as they are expected to be the same.
So they neither live in am_pod or amster_pod.
This is also why the method is static.

File: cicd/forgeops-tests/tests/smoke/am/test_am_metadata.py [hunk 4]
```diff
@@ -2,234 +2,86 @@
...
-                       'revision': config["Amster"]["revision"],
-                       'date': config["Amster"]["date"],
-                       'jdk_version': config["Amster"]["jdk_version"],
-                       'jre_version': config["Amster"]["jre_version"],
-                       'jdk_full_version': config["Amster"]["jdk_full_version"],
-                       'commons_version': config["Amster"]["commons_version"],
-                       'amster_jvm': config["Amster"]["amster_jvm"]}
-    legal_notices_string = config["DEFAULT"]["legal_notices"]
-    legal_notices = set(legal_notices_string.split())
+    MANIFEST_FILEPATH = '/tmp/6.5.0-manifest.txt' # TODO: Currently this hard code blocks this change
```
[the remark this answers is not in the document]
Comment (andrewvinall) [reply cicd/forgeops-tests/tests/smoke/am/test_am_metadata.py]: So the biggest difficulty I have had with these tests is how to input the expected values.  This leads on to how to we obtain the expected values too.  Hence the comment in the code: `# TODO: Currently this hard code blocks this change`  However, the principles are in place.

I am afraid this needs more group discussion.

File: cicd/forgeops-tests/lib/utils/pod.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+from configparser import ConfigParser
+import multiset
+
+# Framework imports
+from utils import logger, kubectl
+
+
+class Pod(object):
+
+    def __init__(self, product_type, name, manifest_filepath):
```
[the remark this answers is not in the document]
Comment (andrewvinall) [reply cicd/forgeops-tests/lib/utils/pod.py]: Please see response to `# TODO: Currently this hard code blocks this change` below.

**Pull request merged**